app/actions/websocket/posts.ts

```diff
@@ -0,0 +1,214 @@
+        const rootPost = await queryPostById(database.database, post.root_id);
+
+        if (!rootPost) {
+            fetchPostById(serverUrl, post.root_id);
+        }
+    }
+
+    const currentChannelId = await queryCurrentChannelId(database.database);
+
+    if (post.channel_id === currentChannelId) {
+        const data = {
+            channelId: post.channel_id,
+            rootId: post.root_id,
+            userId: post.user_id,
+            now: Date.now(),
+        };
+        DeviceEventEmitter.emit(Events.USER_STOP_TYPING, data);
+    }
+
+    const {authors} = await fetchPostAuthors(serverUrl, [post], true);
+    if (authors?.length) {
+        const authorsModels = await database.operator.handleUsers({users: authors, prepareRecordsOnly: true});
+        if (authorsModels.length) {
+            models.push(...authorsModels);
+        }
+    }
+
+    // TODO Thread related functionality: https://mattermost.atlassian.net/browse/MM-41084
+    //const viewingGlobalThreads = getViewingGlobalThreads(state);
+    // const collapsedThreadsEnabled = isCollapsedThreadsEnabled(state);
+    // actions.push(receivedNewPost(post, collapsedThreadsEnabled));
+    if (!shouldIgnorePost(post)) {
+        let markAsViewed = false;
+        let markAsRead = false;
+
+        if (!myChannel.manuallyUnread) {
+            if (
+                post.user_id === currentUserId &&
+                !isSystemMessage(post) &&
+                !isFromWebhook(post)
+            ) {
+                markAsViewed = true;
+                markAsRead = false;
+            } else if ((post.channel_id === currentChannelId)) { // TODO: THREADS && !viewingGlobalThreads) {
+                // Don't mark as read if we're in global threads screen
+                // the currentChannelId still refers to previously viewed channel
+                markAsViewed = false;
+                markAsRead = true;
+            }
+        }
+
+        if (markAsRead) {
+            markChannelAsRead(serverUrl, post.channel_id);
+        } else if (markAsViewed) {
+            const {member: viewedAt} = await markChannelAsViewed(serverUrl, post.channel_id, true);
+            if (viewedAt) {
+                models.push(viewedAt);
+            }
+        } else {
+            const hasMentions = msg.data.mentions.includes(currentUserId);
+            const {member: unreadAt} = await markChannelAsUnread(
+                serverUrl,
+                post.channel_id,
+                myChannel.messageCount + 1,
+                myChannel.mentionsCount + (hasMentions ? 1 : 0),
+                false,
+                myChannel.lastViewedAt,
+                true,
+            );
+            if (unreadAt) {
+                models.push(unreadAt);
+            }
+        }
+    }
+
+    database.operator.batchRecords(models);
+}
+
+export async function handlePostEdited(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
```

Comment:
A small nit...if you extract the operator from here... you would avoid the long chains `await database.operator.handleUsers({`

```suggestion
    const operator = DatabaseManager.serverDatabases[serverUrl].operator;
```